logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/ConversationRepository.kt

```diff
@@ -432,16 +432,20 @@ internal class ConversationDataSource internal constructor(
     ): Either<CoreFailure, Boolean> = wrapStorageRequest {
         val isNewConversation = conversationDAO.getConversationById(conversation.id.toDao()) == null
         if (isNewConversation) {
-            conversationDAO.insertConversation(
-                conversationMapper.fromApiModelToDaoModel(
-                    conversation,
-                    mlsGroupState = conversation.groupId?.let { mlsGroupState(idMapper.fromGroupIDEntity(it), originatedFromEvent) },
-                    selfTeamIdProvider().getOrNull(),
+            val mlsGroupState = conversation.groupId?.let { mlsGroupState(idMapper.fromGroupIDEntity(it), originatedFromEvent) }
+            // if group state is left, then we don't want to persist the conversation
+            if (mlsGroupState?.fold({ true }, { false }) != true) {
```

Comment:
Can't we use `isRight()` instead of fold()?

```suggestion
            if (mlsGroupState?.isRight()
```

Reply:
I don't know why but tests where failing when I was using is left or right